docs/building/forms.md

```diff
@@ -0,0 +1,107 @@
+                    {'type': 'option',
+                     'value': 'low',
+                     'description': 'Low priority'}]],
+ 'header': 'create task',
+ 'footer': None,
+ 'meta': {'confirmation_needed': False},
+ 'method': 'POST',
+ 'path': '/task/create',
+ 'type': 'form'}
+```
+
+Notice the **'type': 'form'** key value pair, which indicates the form response.
+
+So when the user sends **a** to access the **New todo** option item, as mentioned [here](/building/menus/) ONEm will send a GET request to **http://your-callback.url/task/create/**. This url should return something like the above json structure.
+
+The user will be taken through a wizard and it will look like:
+
+```
+#TODO CREATE TASK
+Provide a description
+```
+
+This is the first step and whatever the user replies with will be set as the task description. This is a **string** and it will be **POST**ed to **path** as indicated in the json structure. Notice there is no **header** or **footer** mentioned at this step, so form header and footer will be used as fallbacks.
+
+
+```
+#TODO DUE DATE
+Provide a description
+```
+
+This is the second step and the user needs to reply with a date. There is no callback path so no HTTP request will be performed after this step. The **header** is present here, so it will be displayed instead of the form header.
+
+
+```
+#TODO PRIORITY
+A High priority
+B Low priority
+--Reply A-B
+```
+
+This is the last step and the user needs to choose one option.
+
+After all steps have been processed, as indicated through **method** and **path**, an HTTP POST will be sent to the path which is relative to the callback url: **http://your-callback.url/task/create/** under the mapping **form_item_name=user_choice**
+
+So the POST will look like: **?name=some_description&date=2019-10-10&prio=high**
+
+
+## Type
+The response **type** should be equal to **form** to indicate a form response.
+
+## Meta
+The **meta** key holds a dictionary which can contain one of the following keys:
+
+- **confirmation_needed** - if set to false, there won't be any confirmation screen at the end of the form (defaults to true)
+- **completion_status_show** - if set to true, there will be a completion status shown (defaults to false)
+- **completion_status_in_header** - if set to true, the status will be shown in the header (defaults to false, applicable only when status is shown)
+
+## Header
+The header of the form is indicated through **header** key. This value is not final and will be altered by the ONEm platform, by making it uppercased and placing the name of your app in front of it.
+
+
+## Footer
+The footer of the form is indicated through **footer** key and like the header of the form, it is not final and will be altered by the ONEm platform. If no footer is specified, a default might be set.
+
+## Body
+The body holds a sequence of dictionaries. Each dictoinary is called a form item and it is described through the below keys:
```

Comment:
There is a typo here `dictoinary`. However, I'm not sure if "dictionary" is recognized in JSON language. I think "object" is more appropriate.

Reply:
Yes, object is the javascript naming for the map data structure. Dictionary is widely used as well. As long as we are talking json, which originates from javascript, we could call them objects, yes.